sandbox-tools/merge-repos/src/syncReposToStaging.ts

```diff
@@ -507,100 +509,222 @@ async function resolveConflictsToTheirs(git: SimpleGit, commitDetails: ICommitDe
                 }
+            } else {
+                // Assume file
+                if (validateFile(baseFolder + "/" +  theFile,  destFolder + "/" +  theFile)) {
 
-                if (!moved) {
                     await git.raw([
-                        "mv",
-                        "--force",
-                        "--verbose",
-                        fullInputPath + (isSrcDir ? "/" : ""),
-                        theGitDestFolder + (isSrcDir ? "/" + inputFile + "/" : "")
-                    ]);
-
-                    commitMessage = appendCommitMessage(commitMessage, ` - ${fullInputPath}${isSrcDir ? "/" : ""}`)
+                        "add",
+                        "-f",
+                        destFolder + "/" +  theFile]);
                 }
+            }
+        } else {
+            log(` - (xI) ${baseFolder + "/" + masterFiles[mLp]} ==> ${theFile}`);
+        }
+    }
+
+    // Remove any files / folders that do not exist in the master repo
+    for (let dLp = 0; dLp < destFiles.length; dLp++) {
+        let destFile = destFiles[dLp];
+        if (masterFiles.indexOf(destFile) === -1 && !_isVerifyIgnore(destFile, true)) {
+            let destStats = fs.statSync(destFolder + "/" + destFile);
+            if (destStats.isDirectory()) {
+                commitDetails.message = logAppendMessage(commitDetails.message, { index: "*", working_dir: "F", path: destFolder + "/" + destFile } , `Removing extra folder ${destFile}`);
+                await git.raw([
+                    "rm",
+                    "-f",
+                    "-r",
+                    destFolder + "/" + destFile]);
+                // fs.rmdirSync(destFolder + "/" + destFile, {
+                //     recursive: true
+                // });
             } else {
-                log(` - Ignoring ${inputFile}  (${destFolder})`);
+                commitDetails.message = logAppendMessage(commitDetails.message, { index: "*", working_dir: "E", path: destFolder + "/" + destFile } , "Removing extra file");
+                await git.rm(destFolder + "/" + destFile);
             }
-        };
+        }
+    }
+}
+
+async function getAndSyncSrcRepo(git: SimpleGit, repoName: string, details: IRepoDetails): Promise<ISourceRepoDetails> {
+
+    // Get a clone of the source repo and reset to the starting point
+    let srcOriginRepoUrl = details.url;
+    let srcOriginBranch = details.branch;
+
+    let forkDestOrg =  _mergeGitRoot + "-" + repoName;
+
+    // Now lets go and create a local clone
+    log(`Cloning the source repo ${srcOriginRepoUrl} branch ${srcOriginBranch} to ${forkDestOrg}`);
+    await git.clone(srcOriginRepoUrl, forkDestOrg, [ "-b", srcOriginBranch]);
+
+    let orgGit = createGit(forkDestOrg, "merge.org." + repoName + ".git");
+    let checkoutArgs = [
+        "--progress",
+        "-B", srcOriginBranch
+    ];
 
-        commitDetails.message += commitMessage;
+    if (details.branchStartPoint) {
```

Comment:
Sync to the "start" point (defaults the HEAD) -- this was used for local testing and simulating multiple "update" runs using different points in time.